prime-dotnet-webapi/Models/Enrollee.cs

```diff
@@ -98,10 +101,22 @@ public EnrolmentStatus CurrentStatus
         public ICollection<Status> AvailableStatuses { get; set; }
 
         [NotMapped]
-        public DateTime? AppliedDate { get => this.EnrolmentStatuses?.SingleOrDefault(es => es.StatusCode == Status.SUBMITTED_CODE)?.StatusDate; }
+        public DateTime? AppliedDate
+        {
+            get => this.EnrolmentStatuses?
+                .OrderByDescending(en => en.StatusDate)
+                .FirstOrDefault(es => es.StatusCode == Status.SUBMITTED_CODE)?
+                .StatusDate;
+        }
 
         [NotMapped]
-        public DateTime? ApprovedDate { get => this.EnrolmentStatuses?.SingleOrDefault(es => es.StatusCode == Status.APPROVED_CODE)?.StatusDate; }
+        public DateTime? ApprovedDate
+        {
+            get => this.EnrolmentStatuses?
+                .OrderByDescending(en => en.StatusDate)
+                .FirstOrDefault(es => es.StatusCode == Status.APPROVED_CODE)?
+                .StatusDate;
+        }
 
```

Comment:
Martin and I have devised a more expressive way of doing this, to be added in a later PR (deadline issues)